### ISSUE TITLE
Batch export of liberation data by sample

### ISSUE THREAD

**Issue by Matt-Power** [Bruker-Amics/bugs_features#312]
All versions.
Batch export of liberation data does not extract data as is appears in Process - the data is placed on multiple different tabs. Each is grouped based on the either data column or row. There is no option to group by sample with is a big omission.

There needs to be an option to group by sample (as is done in the Associations batch export).

This applies to many other calculation tables too (e.g. liberation etc).

**Comment by jonathan-knapp** [comment]
I want to be clear I understand this request. It is very different from all other requests we have had regarding batch calculations. You want to have all liberation data for each sample on a different tab. The liberation report for each sample would have all of the liberation classes but would not be able to compare. 

How it is now: 
![image.png](https://images.zenhubusercontent.com/60a274843870252752494569/e4e7047f-4307-4c4b-b0e0-d0d4dae7eae9)

How you want it: 
![image.png](https://images.zenhubusercontent.com/60a274843870252752494569/ca6d47c3-4299-4d4b-9206-2cc8f31711ba)

**Comment by jonathan-knapp** [comment]
@MarkGuo7 to implement this we would have to have the UI of the batch export have a new option of "samples in tabs" under "Sample Arrangement": 
![image.png](https://images.zenhubusercontent.com/60a274843870252752494569/d9de16ee-bbce-4921-820c-8b934a2d5dad)

**Comment by Matt-Power** [comment]
Yes, second example. It is the same way that the associations report now works.

**Comment by Matt-Power** [comment]
In the UI, could you not add a "Group by Sample" in the Data Sheet box?

**Comment by jonathan-knapp** [comment]
@ChiLyMan what is the status of this?

**Comment by ChiLyMan** [comment]
Need to test, @AlexeiDolgolyov has added a bunch of calc exports now

**Comment by ChiLyMan** [comment]
As per Matt's request, Liberation will now have each class exported into a different tab, with all samples shown in each tab.
<img width="1138" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeU42Q0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1a35531f88c31b1df1e40d6a6ba393e097bb86d5/image.png" alt="image.png" />

**Comment by Matt-Power** [comment]
@ChiLyMan Unfortunately, his is not what I requested either - this is how it was previously extracted which is an absolute nightmare to reconstruct for reporting and plotting. What I wanted was the ability to extract each **sample** or an individual tab i.e. each tab is what you see in AMiCS. JK's post from Oct 27 has an example of what I want at the bottom.

This goes for several of the other data tables too - an option for batch export "as-is" or similar where the tables are simply copied sample by sample with each sample pasted into individual tabs. I think this is how the association data are handled.

**Comment by ChiLyMan** [comment]
@Matt-Power OK I thought it was strange you wanted it this way. And those images do not open for me, but I understand what you want now.

**Comment by AlexeiDolgolyov** [comment]
As I understand It should look like Jonathan described in post from Oct 27. Should that work for all type of calculations? (not only Liberation)

<img width="1227" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMWQ5Q0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--403c3846c4bf725f79323db3cac05432e290e086/image.png" alt="image.png" />

**Comment by Matt-Power** [comment]
@AlexeiDolgolyov Yes, I think that it would be useful to have as an option for all calculated data (liberation, size etc) but certainly not default, particularly for calculations such as modal mineralogy where the data fit neatly onto one table (and which is fine the way it is).

As a related point, it would be good to be able to extract just the liberation data itself e.g. the 3 lines of data at top. i.e. Is it possible to include an option to ignore the mineralogy data? The liberation data would then fit on 3 neat tables (one for each of: particle count, distribution, accumulation). Even better though would be to extracted the data to fit on one table / tab with 3 entries per sample number. Left-most column would be sample number, next column class name (e.g. particle count, distribution, accumulation - so each sample takes up 3 rows) and then the class sizes moving left to right as normal. This is usually how I end up formatting and extracting for reports so would save a lot of time for me.

Similar to the above - it would be great to have data on one table for the size distribution calculations (i.e. 3 lines per sample with count, retained, cumulative) and then the P data included in extra columns to the right of the size data columns. The current batch output where the particle, retained, cumulative are all on different tabs is a pain to re-format and re-compile.

**Comment by vfishbain** [comment]
Done. 
This is how the new option looks:
<img width="825" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL0c3Q0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c158cce37fdb0ddb86c183de49e7794dfbe55587/image.png" alt="image.png" />
<img width="832" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL1M3Q0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--8065d96bf4383c17dd9629e25fb6995d53ae4529/image.png" alt="image.png" />
